types/jest/index.d.ts

```diff
@@ -175,67 +177,71 @@ declare namespace jest {
         stringMatching(str: string | RegExp): any;
     }
 
-    interface Matchers {
+    interface Matchers<R> {
```

Comment:
Ping @mhegazy - this seems pretty strange behavior but it doesn't seem unreasonable. are you okay to pull this in?